src/common/exif.cc

```diff
@@ -910,25 +910,6 @@ static void dt_exif_apply_global_overwrites(dt_image_t *img)
   }
 }
 
-// TODO: can this blob also contain xmp and iptc data?
-int dt_exif_read_from_blob(dt_image_t *img, uint8_t *blob, const int size)
-{
-  try
-  {
-    Exiv2::ExifData exifData;
-    Exiv2::ExifParser::decode(exifData, blob, size);
-    bool res = dt_exif_read_exif_data(img, exifData);
-    dt_exif_apply_global_overwrites(img);
```

Comment:
In your new version of the function, `dt_exif_apply_global_overwrites()` is not called.
Since `_dt_exif_read()` calls `dt_exif_apply_global_overwrites()` only after reading exif, but not IPTC and XMP data, it probably changes the output.